File: examples/egress.r
library(jsonlite)
library(httr)
library(readr)

opticquery <- readr::read_file("/Users/dwanczow/Documents/workspace/mesh/examples/query.dsl")
req <-
  httr::POST("http://localhost:8011/v1/rows?column-types=header",
             httr::add_headers(
               "Content-Type" = "application/vnd.marklogic.querydsl+javascript"
             ),
             body = opticquery,
             authenticate('demo-user', 'demo123', type = "digest")
  )


stop_for_status(req)

resp <- content(req, "text")
rows <- jsonlite::fromJSON(resp)$rows
View(rows)

File: examples/egress.py
import requests
from requests.auth import HTTPDigestAuth

import pandas
import json

query = open('query.dsl', 'r').read()

resp = requests.post(
    "http://localhost:8011/v1/rows?column-types=header",
    data=query,
    headers={"Content-Type": "application/vnd.marklogic.querydsl+javascript"},
    auth=HTTPDigestAuth('demo-user', 'demo123'))

data = json.loads(resp.text)
frame = pandas.DataFrame.from_dict(data['rows'])

frame
